### PR TITLE
Implement ECR

### LINKED ISSUE #289: Implement ECR
AWS - ECR

Amazon Elastic Container Registry (ECR) is a fully-managed Docker container registry that makes it easy for developers to store, manage, and deploy Docker container images. Amazon ECR is integrated with Amazon Elastic Container Service (ECS), simplifying your development to production workflow. Amazon ECR eliminates the need to operate your own container repositories or worry about scaling the underlying infrastructure. Amazon ECR hosts your images in a highly available and scalable architecture, allowing you to reliably deploy containers for your applications. Integration with AWS Identity and Access Management (IAM) provides resource-level control of each repository. With Amazon ECR, there are no upfront fees or commitments. You pay only for the amount of data you store in your repositories and data transferred to the Internet. [more info](https://aws.amazon.com/ecr/)

Api: [link](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/ecr/EcrClient.html)

Console: [link](https://us-west-2.console.aws.amazon.com/ecs/home?region=us-west-2#/clusters)

Screen Shot 2020-07-31 at 3 27 45 PM

### PR DESCRIPTION
**Pull request by harjain99**
Fixes #289

### CODE REVIEW

File: examples/ecr/repository.gyro [hunk 1]
```diff
@@ -0,0 +1,15 @@
+aws::ecr-repository example-repo
+    repository-name: "example-repo"
+
+    lifecycle-policy
+        lifecycle-policy: 'lifecycle-policy.json'
+    end
```
Comment (deepanjan90) [inline examples/ecr/repository.gyro:6]: I feel Lifecycle policy resource should be a direct field such that

```
lifecycle-policy: 'lifecycle-policy.json'
```

File: src/main/java/gyro/aws/ecr/EcrRepositoryResource.java [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+import software.amazon.awssdk.services.ecr.model.LifecyclePolicyNotFoundException;
+import software.amazon.awssdk.services.ecr.model.ListTagsForResourceResponse;
+import software.amazon.awssdk.services.ecr.model.Repository;
+import software.amazon.awssdk.services.ecr.model.RepositoryPolicyNotFoundException;
+import software.amazon.awssdk.services.ecr.model.Tag;
+import software.amazon.awssdk.services.ecr.model.TagResourceRequest;
+import software.amazon.awssdk.services.ecr.model.UntagResourceRequest;
+
+/**
+ * Create VPN Gateway.
```
Comment (deepanjan90) [inline src/main/java/gyro/aws/ecr/EcrRepositoryResource.java:50]: ECR ?

File: src/main/java/gyro/aws/ecr/EcrRepositoryResource.java [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+    public void setLifecyclePolicy(EcrLifecyclePolicyResource lifecyclePolicy) {
+        this.lifecyclePolicy = lifecyclePolicy;
+    }
+
+    /**
+     * The policy for the repository.
+     *
+     * @subresource gyro.aws.ecr.EcrRepositoryPolicyResource
+     */
+    public EcrRepositoryPolicyResource getPolicy() {
```
Comment (deepanjan90) [inline src/main/java/gyro/aws/ecr/EcrRepositoryResource.java:156]: Missing @Updatable tag ?

File: src/main/java/gyro/aws/ecr/EcrRepositoryResource.java [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+
+    public void setImageTagMutability(ImageTagMutability imageTagMutability) {
+        this.imageTagMutability = imageTagMutability;
+    }
+
+    /**
+     * The name of the repository.
+     */
+    @Required
+    @Id
```
Comment (deepanjan90) [inline src/main/java/gyro/aws/ecr/EcrRepositoryResource.java:129]: Try and figure out how is the ECR resource used by other resource, and if they use the name of the resource or the arn to reference it.
That will determine if @ID should be part of the name or arn getter

File: src/main/java/gyro/aws/ecr/EcrRepositoryFinder.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.List;
+import java.util.Map;
+
+import gyro.aws.AwsFinder;
+import gyro.core.Type;
+import software.amazon.awssdk.services.ecr.EcrClient;
+import software.amazon.awssdk.services.ecr.model.Repository;
+
+/**
+ * Query repository.
```
Comment (deepanjan90) [inline src/main/java/gyro/aws/ecr/EcrRepositoryFinder.java:28]:
```suggestion
 * Query ECR repository.
```

**Review by deepanjan90 — CHANGES_REQUESTED** [review]
Inline comments

File: src/main/java/gyro/aws/ecr/EcrRepositoryResource.java (hunk 3 above)
Comment (harjain99) [reply src/main/java/gyro/aws/ecr/EcrRepositoryResource.java:156]: `EcrRepositoryPolicyResource` is a sub-resource with its own create/update/delete, so I believe it doesn't need and Updatable tag, right?

File: src/main/java/gyro/aws/ecr/EcrRepositoryResource.java [hunk 6]
```diff
@@ -0,0 +1,359 @@
...
+ *     aws::ecr-repository example-repo
+ *         repository-name: "example-repo"
+ *
+ *         lifecycle-policy
+ *             lifecycle-policy: 'lifecycle-policy.json'
+ *         end
+ *
+ *         policy
+ *             policy: 'repository-policy.json'
+ *         end
```
Comment (deepanjan90) [inline src/main/java/gyro/aws/ecr/EcrRepositoryResource.java:70]: Update the examples

**Review by deepanjan90 — APPROVED** [review]

**Pull request merged**